<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • tximeta</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.19.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.20.0" id="tximeta-1200">tximeta 1.20.0<a class="anchor" aria-label="anchor" href="#tximeta-1200"></a></h2>
<ul><li>Add argument to summarizeToGene(): <code>assignRanges</code> that takes either “range” (default) or “abundant”, and determines the ranges that are attached to the SE (rowRanges). Note that this new argument does not affect the data aggregation at all (counts and abundance are summarized to gene by <code>tximport</code>). The default behavior of summarizeToGene() returns ranges that correspond to the <code>range</code> of the isoforms of the gene, that is the leftmost basepair to the rightmost basepair of any isoform. The non-default “abundant” instead returns the range of the most abundant isoform in the data, averaging over samples. Information about the choice of range is included in <code>mcols</code>
</li>
<li>Added support for piscem-infer: <code>type="piscem"</code> also auto-detected from file ending.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.19.8" id="tximeta-1198">tximeta 1.19.8<a class="anchor" aria-label="anchor" href="#tximeta-1198"></a></h2>
<ul><li>Added support for piscem-infer: <code>type="piscem"</code> also auto-detected from file ending.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.19.6" id="tximeta-1196">tximeta 1.19.6<a class="anchor" aria-label="anchor" href="#tximeta-1196"></a></h2>
<ul><li>Fixed genome build for mouse M26 and higher to GRCm39, thanks to Charlotte Soneson.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.14.0" id="tximeta-1140">tximeta 1.14.0<a class="anchor" aria-label="anchor" href="#tximeta-1140"></a></h2>
<ul><li>Allow GTF specification in linkedTxome to be a serialized GRanges file (a file path to a .rda or .RData file). This bypasses some apparent issue where makeTxDbFromGFF fails while makeTxDbFromGRanges works.</li>
<li>Up to GENCODE 40 (H.s.), M29 (M.m), and Ensembl 106</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.10.0" id="tximeta-1100">tximeta 1.10.0<a class="anchor" aria-label="anchor" href="#tximeta-1100"></a></h2>
<ul><li>Added more tximeta() messaging about specifying the ‘source’ in linkedTxome. Essentially, this triggers GTF processing behavior that users may want to avoid, and so specifying a string other than “Ensembl” may be preferred. Also added note to vignette.</li>
<li>Added note to vignette about alevin import with tximeta where the ‘tgMap’ step requires gene IDs and not gene symbols.</li>
<li>Added hashes for: GENCODE 38 (H.s.), M27 (M.m), and Ensembl 104; GENCODE 37 (H.s.), M26 (M.m), and Ensembl 103; GENCODE 36 and Ensembl 102.</li>
<li>Fixed a bug around AnnotationHub pulldown when using RefSeq as the source.</li>
<li>Fixed a bug where multiple parsed Ensembl GTF TxDb would be added to the BiocFileCache with the same rname.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.8.0" id="tximeta-180">tximeta 1.8.0<a class="anchor" aria-label="anchor" href="#tximeta-180"></a></h2>
<ul><li>Added ‘fromDb’ argument to addIds() to allow IDs to be added from the associated TxDb/EnsDb instead of the org package (which is used by default). Feature suggestion from Kristoffer Vitting-Seerup.</li>
<li>Added function retrieveCDNA() that will download or load a cached version of the transcript sequences used for quantification. Note that the returned sequences are not ordered or matched to the rows of the SummarizedExperiment object. Feature suggestion from Kristoffer Vitting-Seerup.</li>
<li>Added function addCDS() that will add CDS ranges for coding transcripts (and fills in original ranges for non-coding), as well as a ‘coding’ column as a logical indicator. Feature suggestion from Kristoffer Vitting-Seerup.</li>
<li>Added option that environmental variable TXIMETA_HUB_CACHE can be used to set tximeta’s cache location, to avoid prompting the user on the first run of tximeta().</li>
<li>tximeta() will now pull GENCODE TxDb from AnnotationHub when it is listed there (only Homo sapiens are at this point in time). Thanks to Leonardo Collado-Torres for the suggestion!</li>
<li>Now summarizeToGene() will add a column tx_ids, which is a CharacterList of the transcript IDs.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.6.0" id="tximeta-160">tximeta 1.6.0<a class="anchor" aria-label="anchor" href="#tximeta-160"></a></h2>
<ul><li>Added PLOS Computational Biology citation! :-)</li>
<li>Added function <code>splitSE</code> to split one assay of a SummarizedExperiment into multiple assays, each containing features of a given type.</li>
<li>Added a wrapper function makeDGEList() to simplify making a DGEList for use with edgeR. See vignette for example.</li>
<li>tximeta will now make use of EnsDb created and distributed on AnnotationHub, unless useHub=FALSE. Also, a new function retrieveDb() can be called on a SummarizedExperiment to retrieve the underlying TxDb or EnsDb.</li>
<li>tximeta can now use <code>customMetaInfo</code> argument to locate a custom metadata information file such as <code>meta_info.json</code>, which should contain a tage, <code>index_seq_hash</code>, with the SHA-256 hash value of the reference transcripts.</li>
<li>Added <code>markDuplicateTxps</code> argument to add <code>hasDuplicate</code> and <code>duplicates</code> columns to rowData of SummarizedExperiment. One note is that, for efficiency, this argument and <code>cleanDuplicateTxps</code> will now share a duplicates CharacterList that is stored in the BiocFileCache, with the name <code>dups-...</code>. Therefore, if you have previously used <code>cleanDuplicateTxps</code>, you may need to bfcremove() any <code>dups-...</code> entries. Summarization to gene level will keep track of <code>numDupSets</code> per gene which informs about the number of transcripts sets (equivalence classes by transcript sequence content).</li>
<li>If during the indexing step, user didn’t use –gencode for a Gencode transcriptome file, tximeta will deal with this internally now by stripping all characters after the vertical bar <code>|</code>, in order to match long transcript names in the <code>quant.sf</code> files to the correct transcript names in the GTF.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.4.0" id="tximeta-140">tximeta 1.4.0<a class="anchor" aria-label="anchor" href="#tximeta-140"></a></h2>
<ul><li>tximeta will now pull down RefSeq seqinfo, using the dirname() of the GTF location, and assuming some consistency in the structure of the assembly_report.txt that is located in the same directory. Needs more testing though across releases and organisms.</li>
<li>expanded caching of ranges to exons and genes as well. Exons in particular take a long time to build from TxDb, so this saves quite a lot of time.</li>
<li>new ‘addExons’ function will add exons to trancript-level summarized experiments, by replacing transcript GRanges with exon-by-transcript GRangesList. Purposely designed only for transcript-level, see note in ?addExons</li>
<li>tximeta now also caches the transcript ranges themselves, rather than just the TxDb. This shaves extra seconds off the tximeta() call!</li>
<li>add ‘skipSeqinfo’ argument, which avoids attempting to fetch chromosome information (from UCSC) if set to TRUE.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.0" id="tximeta-120">tximeta 1.2.0<a class="anchor" aria-label="anchor" href="#tximeta-120"></a></h2>
<ul><li>Specifying gene=TRUE in addIds() when rows are transcripts will attempt to use a gene_id column to map the IDs. This usually gives a better mapping rate.</li>
<li>Cut off version number from Ensembl names only (not GENCODE)</li>
<li>Added ‘cleanDuplicateTxps’ argument, which does a lot of work for the user: it downloads the FASTA from the source, identifies duplicate transcripts (identical cDNA sequence) then looks to see if transcripts that are in the quantification files, but missing from the GTF, could be renamed from the list of duplicate transcripts such that they would be present in the GTF.</li>
<li>Added coding * non-coding combinations of Ensembl transcriptomes to the hash table. Must be in this order: coding, then non-coding.</li>
<li>Added support for dammit de novo transcriptomes.</li>
<li>Added summarizeToGene as a method, to avoid conflicts with tximport.</li>
<li>Added in Charlotte’s code to split out GENCODE and Ensembl code for generating transcript ranges.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://mikelove.github.io" class="external-link">Michael Love</a>, <a href="https://csoneson.github.io/" class="external-link">Charlotte Soneson</a>, <a href="https://www.peterhickey.org/" class="external-link">Peter Hickey</a>, <a href="https://combine-lab.github.io/" class="external-link">Rob Patro</a>, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

