<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Add IDs to rowRanges of a SummarizedExperiment — addIds • tximeta</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Add IDs to rowRanges of a SummarizedExperiment — addIds"><meta property="og:description" content='For now this function just works with SummarizedExperiments
with Ensembl gene or transcript IDs. See example
of usage in tximeta vignette. For obtaining
multiple matching IDs for each row of the SummarizedExperiment
set multiVals="list". See select for documentation
on use of multiVals.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.19.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Add IDs to rowRanges of a SummarizedExperiment</h1>
    <small class="dont-index">Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/R/addIds.R" class="external-link"><code>R/addIds.R</code></a></small>
    <div class="hidden name"><code>addIds.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For now this function just works with SummarizedExperiments
with Ensembl gene or transcript IDs. See example
of usage in tximeta vignette. For obtaining
multiple matching IDs for each row of the SummarizedExperiment
set <code>multiVals="list"</code>. See <code>select</code> for documentation
on use of <code>multiVals</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addIds</span><span class="op">(</span><span class="va">se</span>, <span class="va">column</span>, fromDb <span class="op">=</span> <span class="cn">FALSE</span>, gene <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>se</dt>
<dd><p>the SummarizedExperiment</p></dd>


<dt>column</dt>
<dd><p>the name of the new ID to add (a <code>column</code> of the org package
database or of the TxDb/EnsDb is <code>fromDb=TRUE</code>)</p></dd>


<dt>fromDb</dt>
<dd><p>logical, whether to use the TxDb/EnsDb that is associated
with <code>se</code>. Default is FALSE, and an org package is used.
Currently only implemented for transcript level (gene=FALSE).
Column names can be viewed with <code>columns(retrieveDb(se))</code></p></dd>


<dt>gene</dt>
<dd><p>logical, whether to map by genes or transcripts (default is FALSE).
if rows are genes, and easily detected as such (ENSG or ENSMUSG), it will
automatically switch to TRUE. if rows are transcripts and <code>gene=TRUE</code>,
then it will try to use a <code>gene_id</code> column to map IDs to <code>column</code></p></dd>


<dt>...</dt>
<dd><p>arguments passed to <code>mapIds</code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a SummarizedExperiment</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="va">tximeta</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # point to a Salmon quantification file:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; dir &lt;- system.file("extdata/salmon_dm", package="tximportData")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; files &lt;- file.path(dir, "SRR1197474", "quant.sf") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; coldata &lt;- data.frame(files, names="SRR1197474", condition="A", stringsAsFactors=FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # normally we would just run the following which would download the appropriate metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # se &lt;- tximeta(coldata)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # for this example, we instead point to a local path where the GTF can be found</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # by making a linkedTxome:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; indexDir &lt;- file.path(dir, "Dm.BDGP6.22.98_salmon-0.14.1")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; fastaFTP &lt;- c("ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet+               "ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; gtfPath &lt;- file.path(dir, "Drosophila_melanogaster.BDGP6.22.98.gtf.gz")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; makeLinkedTxome(indexDir=indexDir, source="LocalEnsembl", organism="Drosophila melanogaster",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet+                 release="98", genome="BDGP6.22", fasta=fastaFTP, gtf=gtfPath, write=FALSE)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> linkedTxome is same as already in bfc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; se &lt;- tximeta(coldata)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> importing quantifications</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> reading in files with read_tsv</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> found matching linked transcriptome:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [ LocalEnsembl - Drosophila melanogaster - release 98 ]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing TxDb created: 2022-01-19 16:44:14</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'BiocGenerics'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:stats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'S4Vectors'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:utils':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     findMatches</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:base':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing transcript ranges created: 2022-01-19 16:44:14</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Warning: the annotation is missing some transcripts that were quantified.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> 5 out of 33706 txps were missing from GTF/GFF but were in the indexed FASTA.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> (This occurs sometimes with Ensembl txps on haplotype chromosomes.)</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> In order to build a ranged SummarizedExperiment, these txps were removed.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> To keep these txps, and to skip adding ranges, use skipMeta=TRUE</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Example missing txps: [FBtr0307759, FBtr0084079, FBtr0084080, ...]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # to clear the entire linkedTxome table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # (don't run unless you want to clear this table!)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfcloc &lt;- getTximetaBFC()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfc &lt;- BiocFileCache(bfcloc)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfcremove(bfc, bfcquery(bfc, "linkedTxomeTbl")$rid)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Dm.eg.db</span><span class="op">)</span>  </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu">addIds</span><span class="op">(</span><span class="va">se</span>, <span class="st">"REFSEQ"</span>, gene<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mapping to new IDs using org.Dm.eg.db</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> if all matching IDs are desired, and '1:many mappings' are reported,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> set multiVals='list' to obtain all the matching IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'select()' returned 1:many mapping between keys and columns</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Love, Charlotte Soneson, Peter Hickey, Rob Patro, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

