<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Make and load linked transcriptomes ("linkedTxome") — linkedTxome • tximeta</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Make and load linked transcriptomes (" linkedtxome><meta property="og:description" content="makeLinkedTxome reads the digest associated with a Salmon
index at indexDir, and links it to key information
about the transcriptome, including the source, organism,
release, and genome (these are custom character strings),
as well as the locations (e.g. local, HTTP, or FTP) for one or more fasta
files and one gtf file. loadLinkedTxome loads this
information from a JSON file. See Details."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.23.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Make and load linked transcriptomes ("linkedTxome")</h1>
    <small class="dont-index">Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/R/linkedTxome.R" class="external-link"><code>R/linkedTxome.R</code></a></small>
    <div class="hidden name"><code>linkedTxome.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>makeLinkedTxome</code> reads the digest associated with a Salmon
index at <code>indexDir</code>, and links it to key information
about the transcriptome, including the <code>source</code>, <code>organism</code>,
<code>release</code>, and <code>genome</code> (these are custom character strings),
as well as the locations (e.g. local, HTTP, or FTP) for one or more <code>fasta</code>
files and one <code>gtf</code> file. <code>loadLinkedTxome</code> loads this
information from a JSON file. See Details.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">makeLinkedTxome</span><span class="op">(</span></span>
<span>  <span class="va">indexDir</span>,</span>
<span>  <span class="va">source</span>,</span>
<span>  <span class="va">organism</span>,</span>
<span>  <span class="va">release</span>,</span>
<span>  <span class="va">genome</span>,</span>
<span>  <span class="va">fasta</span>,</span>
<span>  <span class="va">gtf</span>,</span>
<span>  write <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">jsonFile</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">loadLinkedTxome</span><span class="op">(</span><span class="va">jsonFile</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>indexDir</dt>
<dd><p>the local path to the Salmon index</p></dd>


<dt>source</dt>
<dd><p>the source of transcriptome (e.g. "de-novo").
Note: if you specify "GENCODE" or "Ensembl", this will trigger
behavior by tximeta that may not be desired: e.g. attempts to
download canonical transcriptome data from AnnotationHub
(unless useHub=FALSE when running tximeta) and parsing of
Ensembl GTF using ensembldb (which may fail if the GTF file
has been modified). For transcriptomes that are defined by
local GTF files, it is recommended to use the terms "LocalGENCODE"
or "LocalEnsembl". Setting "LocalEnsembl" will also strip
version numbers from the FASTA transcript IDs to enable matching
with the Ensembl GTF.</p></dd>


<dt>organism</dt>
<dd><p>organism (e.g. "Homo sapiens")</p></dd>


<dt>release</dt>
<dd><p>release number (e.g. "27")</p></dd>


<dt>genome</dt>
<dd><p>genome (e.g. "GRCh38", or "none")</p></dd>


<dt>fasta</dt>
<dd><p>location(s) for the FASTA transcript sequences
(of which the transcripts used to build the index is equal or a subset).
This can be a local path, or an HTTP or FTP URL</p></dd>


<dt>gtf</dt>
<dd><p>location for the GTF/GFF file
(of which the transcripts used to build the index is equal or a subset).
This can be a local path, or an HTTP or FTP URL
While the <code>fasta</code> argument can take a vector of length greater than one
(more than one FASTA file containing transcripts used in indexing),
the <code>gtf</code> argument has to be a single GTF/GFF file.
This can also be a serialized GRanges object (location of a .rds file)
imported with rtracklayer.
If transcripts were added to a standard set of reference transcripts (e.g. fusion genes,
or pathogen transcripts), it is recommended that the tximeta user would manually
add these to the GTF/GFF file, and post the modified GTF/GFF publicly, such as
on Zenodo. This enables consistent annotation and downstream annotation
tasks, such as by <code>summarizeToGene</code>.</p></dd>


<dt>write</dt>
<dd><p>logical, should a JSON file be written out
which documents the transcriptome digest and metadata? (default is TRUE)</p></dd>


<dt>jsonFile</dt>
<dd><p>the path to the json file for the linkedTxome</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>nothing, the function is run for its side effects</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>makeLinkedTxome</code> links the information about the transcriptome
used for quantification in two ways:
1) the function will store a record in tximeta's cache such that
future import of quantification data will automatically access and
parse the GTF as if the transcriptome were one of those automatically
detected by tximeta. Then all features of tximeta (e.g. summarization
to gene, programmatic adding of IDs or metadata) will be available;
2) it will by default write out a JSON file
that can be shared, or posted online, and which can be read by
<code>loadLinkedTxome</code> which will store the information in tximeta's
cache. This should make the full quantification-import pipeline
computationally reproducible / auditable even for transcriptomes
which differ from those provided by references (GENCODE, Ensembl,
RefSeq).</p>
<p>For further details please see the "Linked transcriptomes"
section of the tximeta vignette.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># point to a Salmon quantification file with an additional artificial transcript</span></span></span>
<span class="r-in"><span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/salmon_dm"</span>, package<span class="op">=</span><span class="st">"tximportData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"SRR1197474.plus"</span>, <span class="st">"quant.sf"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>files<span class="op">=</span><span class="va">file</span>, names<span class="op">=</span><span class="st">"SRR1197474"</span>, sample<span class="op">=</span><span class="st">"1"</span>,</span></span>
<span class="r-in"><span>                      stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># now point to the Salmon index itself to create a linkedTxome</span></span></span>
<span class="r-in"><span><span class="co"># as the index will not match a known txome</span></span></span>
<span class="r-in"><span><span class="va">indexDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"Dm.BDGP6.22.98.plus_salmon-0.14.1"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># point to the source FASTA and GTF:</span></span></span>
<span class="r-in"><span><span class="va">fastaFTP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz"</span>,</span></span>
<span class="r-in"><span>              <span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz"</span>,</span></span>
<span class="r-in"><span>              <span class="st">"extra_transcript.fa.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtfPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"Drosophila_melanogaster.BDGP6.22.98.plus.gtf.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># now create a linkedTxome, linking the Salmon index to its FASTA and GTF sources</span></span></span>
<span class="r-in"><span><span class="fu">makeLinkedTxome</span><span class="op">(</span>indexDir<span class="op">=</span><span class="va">indexDir</span>, source<span class="op">=</span><span class="st">"Ensembl"</span>, organism<span class="op">=</span><span class="st">"Drosophila melanogaster"</span>,</span></span>
<span class="r-in"><span>                release<span class="op">=</span><span class="st">"98"</span>, genome<span class="op">=</span><span class="st">"BDGP6.22"</span>, fasta<span class="op">=</span><span class="va">fastaFTP</span>, gtf<span class="op">=</span><span class="va">gtfPath</span>, write<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> NOTE: linkedTxome with source='Ensembl', ensembldb will be used to parse GTF.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> this may produce errors if the GTF is not from Ensembl, or has been modified.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> set useHub=FALSE in tximeta to avoid download of reference txome from AnnotationHub.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> alternatively use a different string for source argument</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> saving linkedTxome in bfc</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to clear the entire linkedTxome table</span></span></span>
<span class="r-in"><span><span class="co"># (don't run unless you want to clear this table!)</span></span></span>
<span class="r-in"><span><span class="co"># bfcloc &lt;- getTximetaBFC()</span></span></span>
<span class="r-in"><span><span class="co"># bfc &lt;- BiocFileCache(bfcloc)</span></span></span>
<span class="r-in"><span><span class="co"># bfcremove(bfc, bfcquery(bfc, "linkedTxomeTbl")$rid)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://mikelove.github.io" class="external-link">Michael Love</a>, <a href="https://csoneson.github.io/" class="external-link">Charlotte Soneson</a>, <a href="https://www.peterhickey.org/" class="external-link">Peter Hickey</a>, <a href="https://combine-lab.github.io/" class="external-link">Rob Patro</a>, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

