<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Summarize estimated quantitites to gene-level — summarizeToGene,SummarizedExperiment-method • tximeta</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Summarize estimated quantitites to gene-level — summarizeToGene,SummarizedExperiment-method"><meta property="og:description" content="Summarizes abundances, counts, lengths, (and inferential
replicates or variance) from transcript- to gene-level.
Transcript IDs are stored as a CharacterList in the mcols
of the output object.
This function operates on SummarizedExperiment objects, and
will automatically access the relevant TxDb (by either finding it
in the BiocFileCache or by building it from an ftp location).
This function uses the tximport package to perform summarization,
where a method is defined that works on simple lists."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.19.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Summarize estimated quantitites to gene-level</h1>
    <small class="dont-index">Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/R/summarizeToGene.R" class="external-link"><code>R/summarizeToGene.R</code></a></small>
    <div class="hidden name"><code>summarizeToGene.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Summarizes abundances, counts, lengths, (and inferential
replicates or variance) from transcript- to gene-level.
Transcript IDs are stored as a CharacterList in the <code>mcols</code>
of the output object.
This function operates on SummarizedExperiment objects, and
will automatically access the relevant TxDb (by either finding it
in the BiocFileCache or by building it from an ftp location).
This function uses the tximport package to perform summarization,
where a method is defined that works on simple lists.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># S4 method for SummarizedExperiment</span></span>
<span><span class="fu">summarizeToGene</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  assignRanges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"range"</span>, <span class="st">"abundant"</span><span class="op">)</span>,</span>
<span>  varReduce <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>a SummarizedExperiment produced by <code>tximeta</code></p></dd>


<dt>assignRanges</dt>
<dd><p><code>"range"</code> or <code>"abundant"</code>, this argument
controls the way that the <code>rowRanges</code> of the output object are assigned
(note that this argument does not affect data aggregation at all).
The default is to just output the entire <code>range</code> of the gene,
i.e. the leftmost basepair to the rightmost basepair across all isoforms.
Alternatively, for expressed genes, one can obtain the start and end
of the most <code>abundant</code> isoform (averaging over all samples).
Non-expressed genes will have <code>range</code>-based positions.
For <code>abundant</code>, for expressed genes,
the name of the range-assigned <code>isoform</code>, <code>max_prop</code>
(maximum isoform proportion), and <code>iso_prop</code> (numeric values for
isoform proportions) are also returned in <code>mcols</code></p></dd>


<dt>varReduce</dt>
<dd><p>whether to reduce per-sample inferential replicates
information into a matrix of sample variances <code>variance</code> (default FALSE)</p></dd>


<dt>...</dt>
<dd><p>arguments passed to <code>tximport</code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a SummarizedExperiment with summarized quantifications
and transcript IDs as a CharacterList in the <code>mcols</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="va">tximeta</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # point to a Salmon quantification file:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; dir &lt;- system.file("extdata/salmon_dm", package="tximportData")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; files &lt;- file.path(dir, "SRR1197474", "quant.sf") </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; coldata &lt;- data.frame(files, names="SRR1197474", condition="A", stringsAsFactors=FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # normally we would just run the following which would download the appropriate metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # se &lt;- tximeta(coldata)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # for this example, we instead point to a local path where the GTF can be found</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # by making a linkedTxome:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; indexDir &lt;- file.path(dir, "Dm.BDGP6.22.98_salmon-0.14.1")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; fastaFTP &lt;- c("ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet+               "ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; gtfPath &lt;- file.path(dir, "Drosophila_melanogaster.BDGP6.22.98.gtf.gz")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; makeLinkedTxome(indexDir=indexDir, source="LocalEnsembl", organism="Drosophila melanogaster",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet+                 release="98", genome="BDGP6.22", fasta=fastaFTP, gtf=gtfPath, write=FALSE)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> linkedTxome is same as already in bfc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; se &lt;- tximeta(coldata)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> importing quantifications</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> reading in files with read.delim (install 'readr' package for speed up)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> found matching linked transcriptome:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [ LocalEnsembl - Drosophila melanogaster - release 98 ]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing TxDb created: 2023-10-09 18:50:37</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing transcript ranges created: 2023-10-09 18:50:37</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Warning: the annotation is missing some transcripts that were quantified.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> 5 out of 33706 txps were missing from GTF/GFF but were in the indexed FASTA.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> (This occurs sometimes with Ensembl txps on haplotype chromosomes.)</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> In order to build a ranged SummarizedExperiment, these txps were removed.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> To keep these txps, and to skip adding ranges, use skipMeta=TRUE</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> Example missing txps: [FBtr0307759, FBtr0084079, FBtr0084080, ...]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # to clear the entire linkedTxome table</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # (don't run unless you want to clear this table!)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfcloc &lt;- getTximetaBFC()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfc &lt;- BiocFileCache(bfcloc)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; # bfcremove(bfc, bfcquery(bfc, "linkedTxomeTbl")$rid)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> tximet&gt; </span>
<span class="r-in"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing TxDb created: 2023-10-09 18:50:37</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> obtaining transcript-to-gene mapping from database</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> generating gene ranges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene ranges assigned by total range of isoforms, see `assignRanges`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing abundance</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing counts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> summarizing length</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://mikelove.github.io" class="external-link">Michael Love</a>, <a href="https://csoneson.github.io/" class="external-link">Charlotte Soneson</a>, <a href="https://www.peterhickey.org/" class="external-link">Peter Hickey</a>, <a href="https://combine-lab.github.io/" class="external-link">Rob Patro</a>, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

