<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import transcript-level quantification data with automatic metadata — tximeta-package • tximeta</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import transcript-level quantification data with automatic metadata — tximeta-package"><meta property="og:description" content="The tximeta package imports abundances (TPM), estimated counts,
and effective lengths from Salmon, alevin, or other quantification
tools, and will output a SummarizedExperiment object. For Salmon
and alevin quantification directories, tximeta will
try to identify the correct provenance of the reference transcripts
and automatically attach the transcript ranges to the
SummarizedExperiment, to facilitate downstream integration with
other datasets. The automatic identification of reference transcripts
should work out-of-the-box for human or mouse transcriptomes from
the sources: GENCODE, Ensembl, or RefSeq."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.19.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import transcript-level quantification data with automatic metadata</h1>
    <small class="dont-index">Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/R/tximeta.R" class="external-link"><code>R/tximeta.R</code></a></small>
    <div class="hidden name"><code>tximeta-package.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The tximeta package imports abundances (TPM), estimated counts,
and effective lengths from Salmon, alevin, or other quantification
tools, and will output a SummarizedExperiment object. For Salmon
and alevin quantification directories, <code>tximeta</code> will
try to identify the correct provenance of the reference transcripts
and automatically attach the transcript ranges to the
SummarizedExperiment, to facilitate downstream integration with
other datasets. The automatic identification of reference transcripts
should work out-of-the-box for human or mouse transcriptomes from
the sources: GENCODE, Ensembl, or RefSeq.</p>
    </div>


    <div id="details">
    <h2>Details</h2>
    <p>The main functions are:</p><ul><li><p><code><a href="tximeta.html">tximeta</a></code> - with key argument: <code>coldata</code></p></li>
<li><p><code>summarizeToGene</code> - summarize quants to gene-level</p></li>
<li><p><code><a href="retrieveDb.html">retrieveDb</a></code> - retrieve the transcript database</p></li>
<li><p><code><a href="addIds.html">addIds</a></code> - add transcript or gene ID (see <code>gene</code> argument)</p></li>
<li><p><code><a href="addExons.html">addExons</a></code> - convert from GRanges to GRangesList</p></li>
</ul><p>All software-related questions should be posted to the Bioconductor Support Site:</p>
<p><a href="https://support.bioconductor.org" class="external-link">https://support.bioconductor.org</a></p>
<p>The code can be viewed at the GitHub repository,
which also lists the contributor code of conduct:</p>
<p><a href="https://github.com/mikelove/tximeta" class="external-link">https://github.com/mikelove/tximeta</a></p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p><strong>tximeta</strong> reference:</p>
<p>Michael I. Love, Charlotte Soneson, Peter F. Hickey, Lisa K. Johnson
N. Tessa Pierce, Lori Shepherd, Martin Morgan, Rob Patro (2020)
Tximeta: reference sequence checksums for provenance identification
in RNA-seq. PLOS Computational Biology.
<a href="https://doi.org/10.1371/journal.pcbi.1007664" class="external-link">https://doi.org/10.1371/journal.pcbi.1007664</a></p>
<p><strong>tximport</strong> reference (the effective length offset and counts-from-abundance):</p>
<p>Charlotte Soneson, Michael I. Love, Mark D. Robinson (2015)
Differential analyses for RNA-seq: transcript-level estimates
improve gene-level inferences. F1000Research.
<a href="http://doi.org/10.12688/f1000research.7563" class="external-link">http://doi.org/10.12688/f1000research.7563</a></p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Michael I. Love, Charlotte Soneson, Peter Hickey, Rob Patro</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Michael Love, Charlotte Soneson, Peter Hickey, Rob Patro, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

