<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Tximeta: transcript quantification import with automatic metadata • tximeta</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Tximeta: transcript quantification import with automatic metadata">
<meta property="og:description" content="tximeta">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tximeta</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.21.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/tximeta.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/thelovelab/tximeta/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Tximeta: transcript quantification import with
automatic metadata</h1>
                        <h4 data-toc-skip class="author">Michael I.
Love, Charlotte Soneson, Peter F. Hickey, Rob Patro</h4>
            
            <h4 data-toc-skip class="date">02/21/2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/thelovelab/tximeta/blob/HEAD/vignettes/tximeta.Rmd" class="external-link"><code>vignettes/tximeta.Rmd</code></a></small>
      <div class="hidden name"><code>tximeta.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Tximeta performs numerous annotation and metadata gathering
      tasks on behalf of users during the import of transcript
      quantifications from <em>Salmon</em>, <em>alevin</em>, or
      <em>piscem-infer</em> into R/Bioconductor. Metadata and transcript
      ranges are added automatically, facilitating genomic analyses and
      assisting in computational reproducibility.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>tximeta</code> package <span class="citation">(Love et al.
2020)</span> extends the <code>tximport</code> package <span class="citation">(Soneson, Love, and Robinson 2015)</span> for import of
transcript-level quantification data into R/Bioconductor. It
automatically adds annotation metadata when the RNA-seq data has been
quantified with <em>Salmon</em> <span class="citation">(Patro et al.
2017)</span> or <a href="https://piscem-infer.readthedocs.io/en/latest/" class="external-link">piscem-infer</a>,
or the scRNA-seq data quantified with <em>alevin</em> <span class="citation">(Srivastava et al. 2019)</span>. To our knowledge,
<code>tximeta</code> is the only package for RNA-seq data import that
can automatically identify and attach transcriptome metadata based on
the unique sequence of the reference transcripts. For more details on
these packages – including the motivation for <code>tximeta</code> and
description of similar work – consult the <strong>References</strong>
below.</p>
<p><strong>Note:</strong> <code>tximeta</code> requires that the
<strong>entire output</strong> of <em>Salmon</em> /
<em>piscem-infer</em> / <em>alevin</em> is present and unmodified in
order to identify the provenance of the reference transcripts. In
general, it’s a good idea to not modify or re-arrange the output
directory of bioinformatic software as other downstream software rely on
and assume a consistent directory structure. For sharing multiple
samples, one can use, for example, <code>tar -czf</code> to bundle up a
set of Salmon output directories, or to bundle one alevin output
directory. For tips on using <code>tximeta</code> with other quantifiers
see the <a href="#other_quantifiers">other quantifiers</a> section
below.</p>
<p><img src="images/diagram.png" width="830"></p>
</div>
<div class="section level2">
<h2 id="tximeta-import-starts-with-sample-table">Tximeta import starts with sample table<a class="anchor" aria-label="anchor" href="#tximeta-import-starts-with-sample-table"></a>
</h2>
<p>The first step using <code>tximeta</code> is to read in the sample
table, which will become the <em>column data</em>, <code>colData</code>,
of the final object, a <em>SummarizedExperiment</em>. The sample table
should contain all the information we need to identify the
<em>Salmon</em> quantification directories. For <em>alevin</em>
quantification, one should point to the <code>quants_mat.gz</code> file
that contains the counts for all of the cells (also, in order to
<code>tximeta</code> to work with <em>alevin</em> quantification, it
requires that <em>alevin</em> was run using gene IDs in the
<code>tgMap</code> step and not gene symbols).</p>
<p>Here we will use a <em>Salmon</em> quantification file in the
<em>tximportData</em> package to demonstrate the usage of
<code>tximeta</code>. We do not have a sample table, so we construct one
in R. It is recommended to keep a sample table as a CSV or TSV file
while working on an RNA-seq project with multiple samples.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/salmon_dm"</span>, package<span class="op">=</span><span class="st">"tximportData"</span><span class="op">)</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"SRR1197474"</span>, <span class="st">"quant.sf"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">files</span>, names<span class="op">=</span><span class="st">"SRR1197474"</span>, condition<span class="op">=</span><span class="st">"A"</span>, stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">coldata</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                   files</span></span>
<span><span class="co">## 1 /__w/_temp/Library/tximportData/extdata/salmon_dm/SRR1197474/quant.sf</span></span>
<span><span class="co">##        names condition</span></span>
<span><span class="co">## 1 SRR1197474         A</span></span></code></pre>
<p><code>tximeta</code> expects at least two columns in
<code>coldata</code>:</p>
<ol style="list-style-type: decimal">
<li>
<code>files</code> - a pointer to the <code>quant.sf</code>
files</li>
<li>
<code>names</code> - the unique names that should be used to
identify samples</li>
</ol>
</div>
<div class="section level2">
<h2 id="running-tximeta">Running tximeta<a class="anchor" aria-label="anchor" href="#running-tximeta"></a>
</h2>
<p>Normally, we would just run <code>tximeta</code> like so:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/tximeta" class="external-link">tximeta</a></span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tximeta.html">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<p>However, to avoid downloading remote GTF files during this vignette,
we will point to a GTF file saved locally (in the <em>tximportData</em>
package). We link the transcriptome of the <em>Salmon</em> index to its
locally saved GTF. The standard recommended usage of
<code>tximeta</code> would be the code chunk above, or to specify a
remote GTF source, not a local one. <strong>This following code is
therefore not recommended for a typically workflow, but is particular to
the vignette code.</strong></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">indexDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"Dm.BDGP6.22.98_salmon-0.14.1"</span><span class="op">)</span></span>
<span><span class="va">fastaFTP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz"</span>,</span>
<span>              <span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz"</span><span class="op">)</span></span>
<span><span class="va">gtfPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>,<span class="st">"Drosophila_melanogaster.BDGP6.22.98.gtf.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/tximeta" class="external-link">tximeta</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/linkedTxome.html">makeLinkedTxome</a></span><span class="op">(</span>indexDir<span class="op">=</span><span class="va">indexDir</span>,</span>
<span>                source<span class="op">=</span><span class="st">"LocalEnsembl"</span>,</span>
<span>                organism<span class="op">=</span><span class="st">"Drosophila melanogaster"</span>,</span>
<span>                release<span class="op">=</span><span class="st">"98"</span>,</span>
<span>                genome<span class="op">=</span><span class="st">"BDGP6.22"</span>,</span>
<span>                fasta<span class="op">=</span><span class="va">fastaFTP</span>,</span>
<span>                gtf<span class="op">=</span><span class="va">gtfPath</span>,</span>
<span>                write<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## saving linkedTxome in bfc (first time)</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/tximeta" class="external-link">tximeta</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tximeta.html">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## importing quantifications</span></span></code></pre>
<pre><code><span><span class="co">## reading in files with read.delim (install 'readr' package for speed up)</span></span></code></pre>
<pre><code><span><span class="co">## 1 </span></span>
<span><span class="co">## found matching linked transcriptome:</span></span>
<span><span class="co">## [ LocalEnsembl - Drosophila melanogaster - release 98 ]</span></span>
<span><span class="co">## building TxDb with 'GenomicFeatures' package</span></span>
<span><span class="co">## Import genomic features from the file as a GRanges object ... OK</span></span>
<span><span class="co">## Prepare the 'metadata' data frame ... OK</span></span>
<span><span class="co">## Make the TxDb object ... OK</span></span>
<span><span class="co">## generating transcript ranges</span></span></code></pre>
<pre><code><span><span class="co">## Warning in checkAssays2Txps(assays, txps): </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Warning: the annotation is missing some transcripts that were quantified.</span></span>
<span><span class="co">## 5 out of 33706 txps were missing from GTF/GFF but were in the indexed FASTA.</span></span>
<span><span class="co">## (This occurs sometimes with Ensembl txps on haplotype chromosomes.)</span></span>
<span><span class="co">## In order to build a ranged SummarizedExperiment, these txps were removed.</span></span>
<span><span class="co">## To keep these txps, and to skip adding ranges, use skipMeta=TRUE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Example missing txps: [FBtr0307759, FBtr0084079, FBtr0084080, ...]</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="what-happened">What happened?<a class="anchor" aria-label="anchor" href="#what-happened"></a>
</h2>
<p><code>tximeta</code> recognized the computed <em>digest</em> of the
transcriptome that the files were quantified against, it accessed the
GTF file of the transcriptome source, found and attached the transcript
ranges, and added the appropriate transcriptome and genome metadata. A
<em>digest</em> is a small string of alphanumeric characters that
uniquely identifies the collection of sequences that were used for
quantification (it is the application of a hash function). We sometimes
also call this value a “checksum” (in the tximeta paper).</p>
<p>A remote GTF is only downloaded once, and a local or remote GTF is
only parsed to build a <em>TxDb</em> or <em>EnsDb</em> once: if
<code>tximeta</code> recognizes that it has seen this <em>Salmon</em>
index before, it will use a cached version of the metadata and
transcript ranges.</p>
<p>Note the warning above that 5 of the transcripts are missing from the
GTF file and so are dropped from the final output. This is a problem
coming from the annotation source, and not easily avoided by
<code>tximeta</code>.</p>
</div>
<div class="section level2">
<h2 id="txdb-ensdb-and-annotationhub">TxDb, EnsDb, and AnnotationHub<a class="anchor" aria-label="anchor" href="#txdb-ensdb-and-annotationhub"></a>
</h2>
<p><code>tximeta</code> makes use of Bioconductor packages for storing
transcript databases as <em>TxDb</em> or <em>EnsDb</em> objects, which
both are connected by default to <code>sqlite</code> backends. For
GENCODE and RefSeq GTF files, <code>tximeta</code> uses the
<em>GenomicFeatures</em> package <span class="citation">(Lawrence
2013)</span> to parse the GTF and build a <em>TxDb</em>. For Ensembl GTF
files, <code>tximeta</code> will first attempt to obtain the correct
<em>EnsDb</em> object using <em>AnnotationHub</em>. The
<em>ensembldb</em> package <span class="citation">(Rainer, Gatto, and
Weichenberger 2019)</span> contains classes and methods for extracting
relevant data from Ensembl files. If the <em>EnsDb</em> has already been
made available on AnnotationHub, <code>tximeta</code> will download the
database directly, which saves the user time parsing the GTF into a
database (to avoid this, set <code>useHub=FALSE</code>). If the relevant
<em>EnsDb</em> is not available on AnnotationHub, <code>tximeta</code>
will build an <em>EnsDb</em> using <em>ensembldb</em> after downloading
the GTF file. Again, the download/construction of a transcript database
occurs only once, and upon subsequent usage of <em>tximeta</em>
functions, the cached version will be used.</p>
</div>
<div class="section level2">
<h2 id="pre-computed-digests">Pre-computed digests<a class="anchor" aria-label="anchor" href="#pre-computed-digests"></a>
</h2>
<p>We plan to support a wide variety of sources and organisms for
transcriptomes with pre-computed digests, though for now the software
focuses on predominantly human and mouse transcriptomes</p>
<p>The following digests are supported in this version of
<code>tximeta</code>:</p>
<table class="table">
<thead><tr class="header">
<th align="left">source</th>
<th align="left">organism</th>
<th align="left">releases</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GENCODE</td>
<td align="left">Homo sapiens</td>
<td align="left">23-45</td>
</tr>
<tr class="even">
<td align="left">GENCODE</td>
<td align="left">Mus musculus</td>
<td align="left">M6-M34</td>
</tr>
<tr class="odd">
<td align="left">Ensembl</td>
<td align="left">Homo sapiens</td>
<td align="left">76-111</td>
</tr>
<tr class="even">
<td align="left">Ensembl</td>
<td align="left">Mus musculus</td>
<td align="left">76-111</td>
</tr>
<tr class="odd">
<td align="left">Ensembl</td>
<td align="left">Drosophila melanogaster</td>
<td align="left">79-111</td>
</tr>
<tr class="even">
<td align="left">RefSeq</td>
<td align="left">Homo sapiens</td>
<td align="left">p1-p13</td>
</tr>
<tr class="odd">
<td align="left">RefSeq</td>
<td align="left">Mus musculus</td>
<td align="left">p2-p6</td>
</tr>
</tbody>
</table>
<p>For Ensembl transcriptomes, we support the combined protein coding
(cDNA) and non-coding (ncRNA) sequences, as well as the protein coding
alone (although the former approach combining coding and non-coding
transcripts is recommended for more accurate quantification).</p>
<p><code>tximeta</code> also has functions to support <em>linked
transcriptomes</em>, where one or more sources for transcript sequences
have been combined or filtered. See the <strong>Linked
transcriptome</strong> section below for a demonstration. (The
<em>makeLinkedTxome</em> function was used above to avoid downloading
the GTF during the vignette building process.)</p>
</div>
<div class="section level2">
<h2 id="summarizedexperiment-output">SummarizedExperiment output<a class="anchor" aria-label="anchor" href="#summarizedexperiment-output"></a>
</h2>
<p>We have our coldata from before. Note that we’ve removed
<code>files</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 1 row and 2 columns</span></span>
<span><span class="co">##                  names   condition</span></span>
<span><span class="co">##            &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## SRR1197474  SRR1197474           A</span></span></code></pre>
<p>Here we show the three matrices that were imported.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "counts"    "abundance" "length"</span></span></code></pre>
<p>If there were inferential replicates (Gibbs samples or bootstrap
samples), these would be imported as additional assays named
<code>"infRep1"</code>, <code>"infRep2"</code>, …</p>
<p><code>tximeta</code> has imported the correct ranges for the
transcripts:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 33701 ranges and 3 metadata columns:</span></span>
<span><span class="co">##               seqnames            ranges strand |     tx_id         gene_id</span></span>
<span><span class="co">##                  &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">##   FBtr0070129        X     656673-657899      + |     28756     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070126        X     656356-657899      + |     28752     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070128        X     656673-657899      + |     28755     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070124        X     656114-657899      + |     28750     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070127        X     656356-657899      + |     28753     FBgn0025637</span></span>
<span><span class="co">##           ...      ...               ...    ... .       ...             ...</span></span>
<span><span class="co">##   FBtr0114299       2R 21325218-21325323      + |      9300     FBgn0086023</span></span>
<span><span class="co">##   FBtr0113582       3R   5598638-5598777      - |     24474     FBgn0082989</span></span>
<span><span class="co">##   FBtr0091635       3L   1488906-1489045      + |     13780     FBgn0086670</span></span>
<span><span class="co">##   FBtr0113599       3L     261803-261953      - |     16973     FBgn0083014</span></span>
<span><span class="co">##   FBtr0113600       3L     831870-832008      - |     17070     FBgn0083057</span></span>
<span><span class="co">##                   tx_name</span></span>
<span><span class="co">##               &lt;character&gt;</span></span>
<span><span class="co">##   FBtr0070129 FBtr0070129</span></span>
<span><span class="co">##   FBtr0070126 FBtr0070126</span></span>
<span><span class="co">##   FBtr0070128 FBtr0070128</span></span>
<span><span class="co">##   FBtr0070124 FBtr0070124</span></span>
<span><span class="co">##   FBtr0070127 FBtr0070127</span></span>
<span><span class="co">##           ...         ...</span></span>
<span><span class="co">##   FBtr0114299 FBtr0114299</span></span>
<span><span class="co">##   FBtr0113582 FBtr0113582</span></span>
<span><span class="co">##   FBtr0091635 FBtr0091635</span></span>
<span><span class="co">##   FBtr0113599 FBtr0113599</span></span>
<span><span class="co">##   FBtr0113600 FBtr0113600</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>We have appropriate genome information, which prevents us from making
bioinformatic mistakes:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Seqinfo object with 25 sequences from an unspecified genome; no seqlengths:</span></span>
<span><span class="co">##   seqnames             seqlengths isCircular genome</span></span>
<span><span class="co">##   2L                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   2R                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   3L                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   3R                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   4                          &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   ...                         ...        ...    ...</span></span>
<span><span class="co">##   211000022280481            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   211000022280494            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   211000022280703            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   mitochondrion_genome       &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   rDNA                       &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="retrieve-the-transcript-database">Retrieve the transcript database<a class="anchor" aria-label="anchor" href="#retrieve-the-transcript-database"></a>
</h2>
<p>The <code>se</code> object has associated metadata that allows
<code>tximeta</code> to link to locally stored cached databases and
other Bioconductor objects. In further sections, we will show examples
functions that leverage this databases for adding exon information,
summarize transcript-level data to the gene level, or add identifiers.
However, first we mention that the user can easily access the cached
database with the following helper function. In this case,
<code>tximeta</code> has an associated <em>EnsDb</em> object that we can
retrieve and use in our R session:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">edb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieveDb.html">retrieveDb</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading existing TxDb created: 2024-02-21 14:31:38</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicFeatures</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">edb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "TxDb"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "GenomicFeatures"</span></span></code></pre>
<p>The database returned by <code>retrieveDb</code> is either a
<em>TxDb</em> in the case of GENCODE or RefSeq GTF annotation file, or
an <em>EnsDb</em> in the case of an Ensembl GTF annotation file. For
further use of these two database objects, consult the
<em>GenomicFeatures</em> vignettes and the <em>ensembldb</em> vignettes,
respectively (both Bioconductor packages).</p>
</div>
<div class="section level2">
<h2 id="add-exons-per-transcript">Add exons per transcript<a class="anchor" aria-label="anchor" href="#add-exons-per-transcript"></a>
</h2>
<p>Because the SummarizedExperiment maintains all the metadata of its
creation, it also keeps a pointer to the necessary database for pulling
out additional information, as demonstrated in the following
sections.</p>
<p>If necessary, the <em>tximeta</em> package can pull down the remote
source to build a TxDb, but given that we’ve already built a TxDb once,
it simply loads the cached version. In order to remove the cached TxDb
and regenerate, one can remove the relevant entry from the
<code>tximeta</code> file cache that resides at the location given by
<code><a href="../reference/getTximetaBFC.html">getTximetaBFC()</a></code>.</p>
<p>The <code>se</code> object created by <code>tximeta</code>, has the
start, end, and strand information for each transcript. Here, we swap
out the transcript <em>GRanges</em> for exons-by-transcript
<em>GRangesList</em> (it is a list of <em>GRanges</em>, where each
element of the list gives the exons for a particular transcript).</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se.exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addExons.html">addExons</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading existing TxDb created: 2024-02-21 14:31:38</span></span></code></pre>
<pre><code><span><span class="co">## generating exon ranges</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">se.exons</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2 ranges and 3 metadata columns:</span></span>
<span><span class="co">##       seqnames        ranges strand |   exon_id      exon_name exon_rank</span></span>
<span><span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;    &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]        X 656673-656740      + |     72949 FBtr0070129-E1         1</span></span>
<span><span class="co">##   [2]        X 657099-657899      + |     72952 FBtr0070129-E2         2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>As with the transcript ranges, the exon ranges will be generated once
and cached locally. As it takes a non-negligible amount of time to
generate the exon-by-transcript <em>GRangesList</em>, this local caching
offers substantial time savings for repeated usage of
<code>addExons</code> with the same transcriptome.</p>
<p>We have implemented <code>addExons</code> to work only on the
transcript-level <em>SummarizedExperiment</em> object. We provide some
motivation for this choice in <code><a href="../reference/addExons.html">?addExons</a></code>. Briefly, if it is
desired to know the exons associated with a particular gene, we feel
that it makes more sense to pull out the relevant set of
exons-by-transcript for the transcripts for this gene, rather than
losing the hierarchical structure (exons to transcripts to genes) that
would occur with a <em>GRangesList</em> of exons grouped per gene.</p>
</div>
<div class="section level2">
<h2 id="easy-summarization-to-gene-level">Easy summarization to gene-level<a class="anchor" aria-label="anchor" href="#easy-summarization-to-gene-level"></a>
</h2>
<p>Likewise, the <em>tximeta</em> package can make use of the cached
TxDb database for the purpose of summarizing transcript-level
quantifications and bias corrections to the gene-level. After
summarization, the <code>rowRanges</code> reflect the start and end
position of the gene, which in Bioconductor are defined by the leftmost
and rightmost genomic coordinates of all the transcripts. As with the
transcript and exons, the gene ranges are cached locally for repeated
usage. The transcript IDs are stored as a <em>CharacterList</em> column
<code>tx_ids</code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading existing TxDb created: 2024-02-21 14:31:38</span></span></code></pre>
<pre><code><span><span class="co">## obtaining transcript-to-gene mapping from database</span></span></code></pre>
<pre><code><span><span class="co">## generating gene ranges</span></span></code></pre>
<pre><code><span><span class="co">## gene ranges assigned by total range of isoforms, see `assignRanges`</span></span></code></pre>
<pre><code><span><span class="co">## summarizing abundance</span></span></code></pre>
<pre><code><span><span class="co">## summarizing counts</span></span></code></pre>
<pre><code><span><span class="co">## summarizing length</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 17208 ranges and 2 metadata columns:</span></span>
<span><span class="co">##               seqnames            ranges strand |     gene_id</span></span>
<span><span class="co">##                  &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##   FBgn0000003       3R   6822498-6822796      + | FBgn0000003</span></span>
<span><span class="co">##   FBgn0000008       2R 22136968-22172834      + | FBgn0000008</span></span>
<span><span class="co">##   FBgn0000014       3R 16807214-16830049      - | FBgn0000014</span></span>
<span><span class="co">##   FBgn0000015       3R 16927212-16972236      - | FBgn0000015</span></span>
<span><span class="co">##   FBgn0000017       3L 16615866-16647882      - | FBgn0000017</span></span>
<span><span class="co">##           ...      ...               ...    ... .         ...</span></span>
<span><span class="co">##   FBgn0286199       3R 24279572-24281576      + | FBgn0286199</span></span>
<span><span class="co">##   FBgn0286203       2R   5413744-5456095      + | FBgn0286203</span></span>
<span><span class="co">##   FBgn0286204       3R   8950246-8963037      - | FBgn0286204</span></span>
<span><span class="co">##   FBgn0286213       3L 13023352-13024762      + | FBgn0286213</span></span>
<span><span class="co">##   FBgn0286222        X   6678424-6681845      + | FBgn0286222</span></span>
<span><span class="co">##                                                tx_ids</span></span>
<span><span class="co">##                                       &lt;CharacterList&gt;</span></span>
<span><span class="co">##   FBgn0000003                             FBtr0081624</span></span>
<span><span class="co">##   FBgn0000008 FBtr0071763,FBtr0100521,FBtr0342981,...</span></span>
<span><span class="co">##   FBgn0000014 FBtr0306337,FBtr0083388,FBtr0083387,...</span></span>
<span><span class="co">##   FBgn0000015 FBtr0415463,FBtr0415464,FBtr0083385,...</span></span>
<span><span class="co">##   FBgn0000017 FBtr0112790,FBtr0345369,FBtr0075357,...</span></span>
<span><span class="co">##           ...                                     ...</span></span>
<span><span class="co">##   FBgn0286199                             FBtr0084600</span></span>
<span><span class="co">##   FBgn0286203 FBtr0299918,FBtr0299920,FBtr0299921,...</span></span>
<span><span class="co">##   FBgn0286204                 FBtr0082014,FBtr0334329</span></span>
<span><span class="co">##   FBgn0286213                             FBtr0075878</span></span>
<span><span class="co">##   FBgn0286222                 FBtr0070953,FBtr0070954</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="section level3">
<h3 id="assign-ranges-by-abundance">Assign ranges by abundance<a class="anchor" aria-label="anchor" href="#assign-ranges-by-abundance"></a>
</h3>
<p>We also offer a new type of range assignment, based on the most
abundant isoform rather than the leftmost to rightmost coordinate. See
the <code>assignRanges</code> argument of <code>?summarizeToGene</code>.
Using the most abundant isoform arguably will reflect more accurate
genomic distances than the default option.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># unevaluated code chunk</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span>, assignRanges<span class="op">=</span><span class="st">"abundant"</span><span class="op">)</span></span></code></pre></div>
<p>For more explanation about why this may be a better choice, see the
following tutorial chapter:</p>
<p><a href="https://tidyomics.github.io/tidy-ranges-tutorial/gene-ranges-in-tximeta.html" class="external-link uri">https://tidyomics.github.io/tidy-ranges-tutorial/gene-ranges-in-tximeta.html</a></p>
<p>In the below diagram, the pink feature is the set of all exons
belonging to any isoform of the gene, such that the TSS is on the right
side of this minus strand feature. However, the blue feature is the most
abundant isoform (the brown features are the next most abundant
isoforms). The pink feature is therefore not a good representation for
the locus.</p>
<p><img src="images/assignRanges-abundant.png" width="672"></p>
</div>
</div>
<div class="section level2">
<h2 id="add-different-identifiers">Add different identifiers<a class="anchor" aria-label="anchor" href="#add-different-identifiers"></a>
</h2>
<p>We would like to add support to easily map transcript or gene
identifiers from one annotation to another. This is just a prototype
function, but we show how we can easily add alternate IDs given that we
know the organism and the source of the transcriptome. (This function
currently only works for GENCODE and Ensembl gene or transcript IDs but
could be extended to work for arbitrary sources.)</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Dm.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addIds.html">addIds</a></span><span class="op">(</span><span class="va">gse</span>, <span class="st">"REFSEQ"</span>, gene<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## mapping to new IDs using org.Dm.eg.db</span></span>
<span><span class="co">## if all matching IDs are desired, and '1:many mappings' are reported,</span></span>
<span><span class="co">## set multiVals='list' to obtain all the matching IDs</span></span></code></pre>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mcols</span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 17208 rows and 3 columns</span></span>
<span><span class="co">##                 gene_id                                  tx_ids       REFSEQ</span></span>
<span><span class="co">##             &lt;character&gt;                         &lt;CharacterList&gt;  &lt;character&gt;</span></span>
<span><span class="co">## FBgn0000003 FBgn0000003                             FBtr0081624    NR_001992</span></span>
<span><span class="co">## FBgn0000008 FBgn0000008 FBtr0071763,FBtr0100521,FBtr0342981,... NM_001014543</span></span>
<span><span class="co">## FBgn0000014 FBgn0000014 FBtr0306337,FBtr0083388,FBtr0083387,... NM_001170161</span></span>
<span><span class="co">## FBgn0000015 FBgn0000015 FBtr0415463,FBtr0415464,FBtr0083385,... NM_001275719</span></span>
<span><span class="co">## FBgn0000017 FBgn0000017 FBtr0112790,FBtr0345369,FBtr0075357,... NM_001104153</span></span>
<span><span class="co">## ...                 ...                                     ...          ...</span></span>
<span><span class="co">## FBgn0286199 FBgn0286199                             FBtr0084600    NM_142982</span></span>
<span><span class="co">## FBgn0286203 FBgn0286203 FBtr0299918,FBtr0299920,FBtr0299921,... NM_001144134</span></span>
<span><span class="co">## FBgn0286204 FBgn0286204                 FBtr0082014,FBtr0334329 NM_001275464</span></span>
<span><span class="co">## FBgn0286213 FBgn0286213                             FBtr0075878    NM_168534</span></span>
<span><span class="co">## FBgn0286222 FBgn0286222                 FBtr0070953,FBtr0070954    NM_132111</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="differential-expression-analysis">Differential expression analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<p>The following code chunk demonstrates how to build a
<em>DESeqDataSet</em> and begin a differential expression analysis.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># here there is a single sample so we use ~1.</span></span>
<span><span class="co"># expect a warning that there is only a single sample...</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">gse</span>, <span class="op">~</span><span class="fl">1</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## using counts and average transcript lengths from tximeta</span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ... see DESeq2 vignette</span></span></code></pre></div>
<p>The <em>Swish</em> method in the <em>fishpond</em> package directly
works with the <em>SummarizedExperiment</em> output from
<em>tximeta</em>, and can perform differential analysis on transcript
expression taking into account inferential replicates, e.g. bootstrap or
Gibbs samples, which are imported and arranged by <code>tximeta</code>
if these were generated during quantification.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://thelovelab.github.io/fishpond" class="external-link">fishpond</a></span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">se</span></span>
<span><span class="co"># y &lt;- scaleInfReps(y)</span></span>
<span><span class="co"># y &lt;- labelKeep(y)</span></span>
<span><span class="co"># y &lt;- swish(y, x="condition")</span></span>
<span><span class="co"># ... see Swish vignette in fishpond package</span></span></code></pre></div>
<p>We have a convenient wrapper function that will build a
<em>DGEList</em> object for use with <em>edgeR</em>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeDGEList.html">makeDGEList</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></span>
<span><span class="co"># ... see edgeR User's Guide for further steps</span></span></code></pre></div>
<p>The following code chunk demonstrates the code inside of the above
wrapper function, and produces the same output.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">[[</span><span class="st">"counts"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">normMat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">[[</span><span class="st">"length"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">normMat</span> <span class="op">&lt;-</span> <span class="va">normMat</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowMeans.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">normMat</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">o</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">cts</span><span class="op">/</span><span class="va">normMat</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">cts</span><span class="op">/</span><span class="va">normMat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="va">cts</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/scaleOffset.html" class="external-link">scaleOffset</a></span><span class="op">(</span><span class="va">y</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">normMat</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">o</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># ... see edgeR User's Guide for further steps</span></span></code></pre></div>
<p>The following code chunk demonstrates how one could use the
<em>Swish</em> method in the fishpond Bioconductor package. Here we use
the transcript-level object <code>se</code>. This dataset only has a
single sample and no inferential replicates, but the analysis would
begin with such code. See the Swish vignette in the fishpond package for
a complete example:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">se</span> <span class="co"># rename the object to 'y'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://thelovelab.github.io/fishpond" class="external-link">fishpond</a></span><span class="op">)</span></span>
<span><span class="co"># if inferential replicates existed in the data,</span></span>
<span><span class="co"># analysis would begin with:</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># y &lt;- scaleInfReps(y)</span></span>
<span><span class="co"># ... see Swish vignette in the fishpond package</span></span></code></pre></div>
<p>For <em>limma</em> with <em>voom</em> transformation we recommend, as
in the <em>tximport</em> vignette to generate counts-from-abundance
instead of providing an offset for average transcript length.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">se</span>, countsFromAbundance<span class="op">=</span><span class="st">"lengthScaledTPM"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## loading existing TxDb created: 2024-02-21 14:31:38</span></span></code></pre>
<pre><code><span><span class="co">## obtaining transcript-to-gene mapping from database</span></span></code></pre>
<pre><code><span><span class="co">## loading existing gene ranges created: 2024-02-21 14:31:42</span></span></code></pre>
<pre><code><span><span class="co">## gene ranges assigned by total range of isoforms, see `assignRanges`</span></span></code></pre>
<pre><code><span><span class="co">## summarizing abundance</span></span></code></pre>
<pre><code><span><span class="co">## summarizing counts</span></span></code></pre>
<pre><code><span><span class="co">## summarizing length</span></span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">[[</span><span class="st">"counts"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co"># see limma User's Guide for further steps</span></span></code></pre></div>
<p>Above we generated counts-from-abundance when calling
<code>summarizeToGene</code>. The counts-from-abundance status is then
stored in the metadata:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">$</span><span class="va">countsFromAbundance</span> </span></code></pre></div>
<pre><code><span><span class="co">## [1] "lengthScaledTPM"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="additional-metadata">Additional metadata<a class="anchor" aria-label="anchor" href="#additional-metadata"></a>
</h2>
<p>The following information is attached to the
<em>SummarizedExperiment</em> by <code>tximeta</code>:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "tximetaInfo"         "quantInfo"           "countsFromAbundance"</span></span>
<span><span class="co">## [4] "level"               "txomeInfo"           "txdbInfo"</span></span></code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[[</span><span class="st">"quantInfo"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 31</span></span>
<span><span class="co">##  $ salmon_version                                        : chr "0.14.1"</span></span>
<span><span class="co">##  $ samp_type                                             : chr "none"</span></span>
<span><span class="co">##  $ opt_type                                              : chr "vb"</span></span>
<span><span class="co">##  $ quant_errors                                          :List of 1</span></span>
<span><span class="co">##   ..$ : list()</span></span>
<span><span class="co">##  $ num_libraries                                         : int 1</span></span>
<span><span class="co">##  $ library_types                                         : chr "ISR"</span></span>
<span><span class="co">##  $ frag_dist_length                                      : int 1001</span></span>
<span><span class="co">##  $ seq_bias_correct                                      : logi FALSE</span></span>
<span><span class="co">##  $ gc_bias_correct                                       : logi TRUE</span></span>
<span><span class="co">##  $ num_bias_bins                                         : int 4096</span></span>
<span><span class="co">##  $ mapping_type                                          : chr "mapping"</span></span>
<span><span class="co">##  $ num_valid_targets                                     : int 33706</span></span>
<span><span class="co">##  $ num_decoy_targets                                     : int 0</span></span>
<span><span class="co">##  $ num_eq_classes                                        : int 70718</span></span>
<span><span class="co">##  $ serialized_eq_classes                                 : logi FALSE</span></span>
<span><span class="co">##  $ length_classes                                        : int [1:5, 1] 867 1533 2379 3854 71382</span></span>
<span><span class="co">##  $ index_seq_hash                                        : chr "7ba5e9597796ea86cf11ccf6635ca88fbc37c2848d38083c23986aa2c6a21eae"</span></span>
<span><span class="co">##  $ index_name_hash                                       : chr "b6426061057bba9b7afb4dc76fa68238414cf35b4190c95ca6fc44280d4ca87c"</span></span>
<span><span class="co">##  $ index_seq_hash512                                     : chr "05f111abcda1efd2e489ace6324128cdaaa311712a28ed716d957fdfd8706ec41ca9177ebf12f54e99c2a89582d06f31c5e09dc1dce2d13"| __truncated__</span></span>
<span><span class="co">##  $ index_name_hash512                                    : chr "ccdf58f23e48c8c53cd122b5f5990b5adce9fec87ddf8bd88153afbe93296d87b818fba89d12dbc20c882f7d98353840394c5040fea7432"| __truncated__</span></span>
<span><span class="co">##  $ num_bootstraps                                        : int 0</span></span>
<span><span class="co">##  $ num_processed                                         : int 42422337</span></span>
<span><span class="co">##  $ num_mapped                                            : int 34098209</span></span>
<span><span class="co">##  $ num_decoy_fragments                                   : int 0</span></span>
<span><span class="co">##  $ num_dovetail_fragments                                : int 2048810</span></span>
<span><span class="co">##  $ num_fragments_filtered_vm                             : int 989383</span></span>
<span><span class="co">##  $ num_alignments_below_threshold_for_mapped_fragments_vm: int 267540</span></span>
<span><span class="co">##  $ percent_mapped                                        : num 80.4</span></span>
<span><span class="co">##  $ call                                                  : chr "quant"</span></span>
<span><span class="co">##  $ start_time                                            : chr "Sat Oct 12 13:55:01 2019"</span></span>
<span><span class="co">##  $ end_time                                              : chr "Sat Oct 12 14:08:11 2019"</span></span></code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[[</span><span class="st">"txomeInfo"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 9</span></span>
<span><span class="co">##  $ index      : chr "Dm.BDGP6.22.98_salmon-0.14.1"</span></span>
<span><span class="co">##  $ source     : chr "LocalEnsembl"</span></span>
<span><span class="co">##  $ organism   : chr "Drosophila melanogaster"</span></span>
<span><span class="co">##  $ release    : chr "98"</span></span>
<span><span class="co">##  $ genome     : chr "BDGP6.22"</span></span>
<span><span class="co">##  $ fasta      :List of 1</span></span>
<span><span class="co">##   ..$ : chr [1:2] "ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz" "ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz"</span></span>
<span><span class="co">##  $ gtf        : chr "/__w/_temp/Library/tximportData/extdata/salmon_dm/Drosophila_melanogaster.BDGP6.22.98.gtf.gz"</span></span>
<span><span class="co">##  $ sha256     : chr "7ba5e9597796ea86cf11ccf6635ca88fbc37c2848d38083c23986aa2c6a21eae"</span></span>
<span><span class="co">##  $ linkedTxome: logi TRUE</span></span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[[</span><span class="st">"tximetaInfo"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 3</span></span>
<span><span class="co">##  $ version   :Classes 'package_version', 'numeric_version'  hidden list of 1</span></span>
<span><span class="co">##   ..$ : int [1:3] 1 21 4</span></span>
<span><span class="co">##  $ type      : chr "salmon"</span></span>
<span><span class="co">##  $ importTime: POSIXct[1:1], format: "2024-02-21 14:31:29"</span></span></code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">[[</span><span class="st">"txdbInfo"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  Named chr [1:13] "TxDb" "GenomicFeatures" ...</span></span>
<span><span class="co">##  - attr(*, "names")= chr [1:13] "Db type" "Supporting package" "Data source" "Organism" ...</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="errors-connecting-to-a-database">Errors connecting to a database<a class="anchor" aria-label="anchor" href="#errors-connecting-to-a-database"></a>
</h2>
<p><code>tximeta</code> makes use of <em>BiocFileCache</em> to store
transcript and other databases, so saving the relevant databases in a
centralized location used by other Bioconductor packages as well. It is
possible that an error can occur in connecting to these databases,
either if the files were accidentally removed from the file system, or
if there was an error generating or writing the database to the cache
location. In each of these cases, it is easy to remove the entry in the
<em>BiocFileCache</em> so that <code>tximeta</code> will know to
regenerate the transcript database or any other missing database.</p>
<p>If you have used the default cache location, then you can obtain
access to your BiocFileCache with:</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocFileCache</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: dbplyr</span></span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Otherwise, you can recall your particular <code>tximeta</code> cache
location with <code><a href="../reference/getTximetaBFC.html">getTximetaBFC()</a></code>.</p>
<p>You can then inspect the entries in your BiocFileCache using
<code>bfcinfo</code> and remove the entry associated with the missing
database with <code>bfcremove</code>. See the BiocFileCache vignette for
more details on finding and removing entries from a BiocFileCache.</p>
<p>Note that there may be many entries in the BiocFileCache location,
including <code>.sqlite</code> database files and serialized
<code>.rds</code> files. You should only remove the entry associated
with the missing database, e.g. if R gave an error when trying to
connect to the TxDb associated with GENCODE v99 human transcripts, you
should look for the <code>rid</code> of the entry associated with the
human v99 GTF from GENCODE.</p>
</div>
<div class="section level2">
<h2 id="what-if-digest-isnt-known">What if digest isn’t known?<a class="anchor" aria-label="anchor" href="#what-if-digest-isnt-known"></a>
</h2>
<p><code>tximeta</code> automatically imports relevant metadata when the
transcriptome matches a known source – <em>known</em> in the sense that
it is in the set of pre-computed hashed digests in <code>tximeta</code>
(GENCODE, Ensembl, and RefSeq for human and mouse). <code>tximeta</code>
also facilitates the linking of transcriptomes used in building the
<em>Salmon</em> index with relevant public sources, in the case that
these are not part of this pre-computed set known to
<code>tximeta</code>. The linking of the transcriptome source with the
quantification files is important in the case that the transcript
sequence no longer matches a known source (uniquely combined or filtered
FASTA files), or if the source is not known to <code>tximeta</code>.
Combinations of coding and non-coding human, mouse, and fruit fly
<em>Ensembl</em> transcripts should be automatically recognized by
<code>tximeta</code> and does not require making a <em>linkedTxome</em>.
As the package is further developed, we plan to roll out support for all
common transcriptomes, from all sources.</p>
<p><strong>Note:</strong> if you are using Salmon in alignment mode,
then there is no Salmon index, and without the Salmon index, there is no
digest. We don’t have a perfect solution for this yet, but you can still
summarize transcript counts to gene with a <code>tx2gene</code> table
that you construct manually (see <code>tximport</code> vignette for
example code). Just specify the arguments,
<code>skipMeta=TRUE, txOut=FALSE, tx2gene=tx2gene</code>, when calling
<code>tximeta</code> and it will perform summarization to gene level as
in <code>tximport</code>.</p>
<p>We now demonstrate how to make a <em>linkedTxome</em> and how to
share and load a <em>linkedTxome</em>. We point to a <em>Salmon</em>
quantification file which was quantified against a transcriptome that
included the coding and non-coding <em>Drosophila melanogaster</em>
transcripts, as well as an artificial transcript of 960 bp (for
demonstration purposes only).</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"SRR1197474.plus"</span>, <span class="st">"quant.sf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>files<span class="op">=</span><span class="va">file</span>, names<span class="op">=</span><span class="st">"SRR1197474"</span>, sample<span class="op">=</span><span class="st">"1"</span>,</span>
<span>                      stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Trying to import the files gives a message that <code>tximeta</code>
couldn’t find a matching transcriptome, so it returns an non-ranged
<em>SummarizedExperiment</em>.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tximeta.html">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## importing quantifications</span></span></code></pre>
<pre><code><span><span class="co">## reading in files with read.delim (install 'readr' package for speed up)</span></span></code></pre>
<pre><code><span><span class="co">## 1 </span></span>
<span><span class="co">## couldn't find matching transcriptome, returning non-ranged SummarizedExperiment</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="linked-transcriptomes">Linked transcriptomes<a class="anchor" aria-label="anchor" href="#linked-transcriptomes"></a>
</h2>
<p>If the transcriptome used to generate the <em>Salmon</em> index does
not match any transcriptomes from known sources (e.g. from combining or
filtering known transcriptome files), there is not much that can be done
to automatically populate the metadata during quantification import.
However, we can facilitate the following two cases:</p>
<ol style="list-style-type: decimal">
<li>the transcriptome was created locally and has been linked to its
public source(s)</li>
<li>the transcriptome was produced by another group, and they have
produced and shared a file that links the transcriptome to public
source(s)</li>
</ol>
<p><code>tximeta</code> offers functionality to assist reproducible
analysis in both of these cases.</p>
<p>To make this quantification reproducible, we make a
<code>linkedTxome</code> which records key information about the sources
of the transcript FASTA files, and the location of the relevant GTF
file. It also records the digest of the transcriptome that was computed
by <em>Salmon</em> during the <code>index</code> step.</p>
<p><strong>Source:</strong> when creating the <code>linkedTxome</code>
one must specify the <code>source</code> of the transcriptome. See
<code><a href="../reference/linkedTxome.html">?linkedTxome</a></code> for a note on the implications of this text
string. For canonical GENCODE or Ensembl transcriptomes, one can use
<code>"GENCODE"</code> or <code>"Ensembl"</code>, but for modified or
otherwise any transcriptomes defined by a local database, it is
recommended to use a different string, <code>"LocalGENCODE"</code> or
`“LocalEnsembl”, which will avoid <em>tximeta</em> pulling canonical
GENCODE or Ensembl resources from AnnotationHub.</p>
<p><strong>Multiple GTF/GFF files:</strong> <code>linkedTxome</code> and
<code>tximeta</code> do not currently support multiple GTF/GFF files,
which is a more complicated case than multiple FASTA, which is
supported. Currently, we recommend that users should add or combine
GTF/GFF files themselves to create a single GTF/GFF file that contains
all features used in quantification, and then upload such a file to
<em>Zenodo</em>, which can then be linked as shown below. Feel free to
contact the developers on the Bioconductor support site or GitHub Issue
page for further details or feature requests.</p>
<p><strong>Stringtie:</strong> A special note for building on top of
Stringtie-generated transcripts: it is a good idea to change gene
identifiers, to <em>not</em> include a period <code>.</code>, as the
period will later be used to separate transcript versions from gene
identifiers. This can be done before building the Salmon index, by
changing periods in the gene identifier to an underscore. See <a href="https://github.com/thelovelab/tximeta/issues/68" class="external-link">this GitHub
issue</a> for details.</p>
<p>By default, <code>linkedTxome</code> will write out a JSON file which
can be shared with others, linking the digest of the index with the
other metadata, including FASTA and GTF sources. By default, it will
write out to a file with the same name as the <code>indexDir</code>, but
with a <code>.json</code> extension added. This can be prevented with
<code>write=FALSE</code>, and the file location can be changed with
<code>jsonFile</code>.</p>
<p>First we specify the path where the <em>Salmon</em> index is
located.</p>
<p>Typically you would not use <code>system.file</code> and
<code>file.path</code> to locate this directory, but simply define
<code>indexDir</code> to be the path of the <em>Salmon</em> directory on
your machine. Here we use <code>system.file</code> and
<code>file.path</code> because we have included parts of a
<em>Salmon</em> index directory in the <em>tximeta</em> package itself
for demonstration of functionality in this vignette.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">indexDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"Dm.BDGP6.22.98.plus_salmon-0.14.1"</span><span class="op">)</span></span></code></pre></div>
<p>Now we provide the location of the FASTA files and the GTF file for
this transcriptome.</p>
<p><strong>Note:</strong> the basename for the GTF file is used as a
unique identifier for the cached versions of the <em>TxDb</em> and the
transcript ranges, which are stored on the user’s behalf via
<em>BiocFileCache</em>. This is not an issue, as GENCODE, Ensembl, and
RefSeq all provide GTF files which are uniquely identified by their
filename,
e.g. <code>Drosophila_melanogaster.BDGP6.22.98.gtf.gz</code>.</p>
<p>The recommended usage of <code>tximeta</code> would be to specify a
remote GTF source, as seen in the commented-out line below:</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fastaFTP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz"</span>,</span>
<span>              <span class="st">"ftp://ftp.ensembl.org/pub/release-98/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz"</span>,</span>
<span>              <span class="st">"extra_transcript.fa.gz"</span><span class="op">)</span></span>
<span><span class="co">#gtfFTP &lt;- "ftp://path/to/custom/Drosophila_melanogaster.BDGP6.22.98.plus.gtf.gz"</span></span></code></pre></div>
<p>Instead of the above commented-out FTP location for the GTF file, we
specify a location within an R package. This step is just to avoid
downloading from a remote FTP during vignette building. This use of
<code>file.path</code> to point to a file in an R package is specific to
this vignette and should not be used in a typical workflow. The
following GTF file is a modified version of the release 98 from Ensembl,
which includes description of a one transcript, one exon artificial gene
which was inserted into the transcriptome (for demonstration purposes
only).</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gtfPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>,<span class="st">"Drosophila_melanogaster.BDGP6.22.98.plus.gtf.gz"</span><span class="op">)</span></span></code></pre></div>
<p>Finally, we create a <em>linkedTxome</em>. In this vignette, we point
to a temporary directory for the JSON file, but a more typical workflow
would write the JSON file to the same location as the <em>Salmon</em>
index by not specifying <code>jsonFile</code>.</p>
<p><code>makeLinkedTxome</code> performs two operation: (1) it creates a
new entry in an internal table that links the transcriptome used in the
<em>Salmon</em> index to its sources, and (2) it creates a JSON file
such that this <em>linkedTxome</em> can be shared.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">jsonFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">indexDir</span><span class="op">)</span>, <span class="st">".json"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/linkedTxome.html">makeLinkedTxome</a></span><span class="op">(</span>indexDir<span class="op">=</span><span class="va">indexDir</span>,</span>
<span>                source<span class="op">=</span><span class="st">"LocalEnsembl"</span>, organism<span class="op">=</span><span class="st">"Drosophila melanogaster"</span>,</span>
<span>                release<span class="op">=</span><span class="st">"98"</span>, genome<span class="op">=</span><span class="st">"BDGP6.22"</span>,</span>
<span>                fasta<span class="op">=</span><span class="va">fastaFTP</span>, gtf<span class="op">=</span><span class="va">gtfPath</span>,</span>
<span>                jsonFile<span class="op">=</span><span class="va">jsonFile</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## writing linkedTxome to /tmp/Rtmph0AffD/Dm.BDGP6.22.98.plus_salmon-0.14.1.json</span></span></code></pre>
<pre><code><span><span class="co">## saving linkedTxome in bfc</span></span></code></pre>
<p>After running <code>makeLinkedTxome</code>, the connection between
this <em>Salmon</em> index (and its digest) with the sources is saved
for persistent usage. Note that because we added a single transcript of
960bp to the FASTA file used for quantification, <code>tximeta</code>
could tell that this was not quantified against release 98 of the
Ensembl transcripts for <em>Drosophila melanogaster</em>. Only when the
correct set of transcripts were specified does <code>tximeta</code>
recognize and import the correct metadata.</p>
<p>With use of <code>tximeta</code> and a <em>linkedTxome</em>, the
software figures out if the remote GTF has been accessed and compiled
into a <em>TxDb</em> before, and on future calls, it will simply load
the pre-computed metadata and transcript ranges.</p>
<p>Note the warning that 5 of the transcripts are missing from the GTF
file and so are dropped from the final output. This is a problem coming
from the annotation source, and not easily avoided by
<code>tximeta</code>.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tximeta.html">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## importing quantifications</span></span></code></pre>
<pre><code><span><span class="co">## reading in files with read.delim (install 'readr' package for speed up)</span></span></code></pre>
<pre><code><span><span class="co">## 1 </span></span>
<span><span class="co">## found matching linked transcriptome:</span></span>
<span><span class="co">## [ LocalEnsembl - Drosophila melanogaster - release 98 ]</span></span>
<span><span class="co">## building TxDb with 'GenomicFeatures' package</span></span>
<span><span class="co">## Import genomic features from the file as a GRanges object ... OK</span></span>
<span><span class="co">## Prepare the 'metadata' data frame ... OK</span></span>
<span><span class="co">## Make the TxDb object ... OK</span></span>
<span><span class="co">## generating transcript ranges</span></span></code></pre>
<pre><code><span><span class="co">## Warning in checkAssays2Txps(assays, txps): </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Warning: the annotation is missing some transcripts that were quantified.</span></span>
<span><span class="co">## 5 out of 33707 txps were missing from GTF/GFF but were in the indexed FASTA.</span></span>
<span><span class="co">## (This occurs sometimes with Ensembl txps on haplotype chromosomes.)</span></span>
<span><span class="co">## In order to build a ranged SummarizedExperiment, these txps were removed.</span></span>
<span><span class="co">## To keep these txps, and to skip adding ranges, use skipMeta=TRUE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Example missing txps: [FBtr0307759, FBtr0084079, FBtr0084080, ...]</span></span></code></pre>
<p>We can see that the appropriate metadata and transcript ranges are
attached.</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 33702 ranges and 3 metadata columns:</span></span>
<span><span class="co">##               seqnames            ranges strand |     tx_id         gene_id</span></span>
<span><span class="co">##                  &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;CharacterList&gt;</span></span>
<span><span class="co">##       Newgene       3R             1-960      + |     20219         Newgene</span></span>
<span><span class="co">##   FBtr0070129        X     656673-657899      + |     28757     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070126        X     656356-657899      + |     28753     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070128        X     656673-657899      + |     28756     FBgn0025637</span></span>
<span><span class="co">##   FBtr0070124        X     656114-657899      + |     28751     FBgn0025637</span></span>
<span><span class="co">##           ...      ...               ...    ... .       ...             ...</span></span>
<span><span class="co">##   FBtr0114299       2R 21325218-21325323      + |      9300     FBgn0086023</span></span>
<span><span class="co">##   FBtr0113582       3R   5598638-5598777      - |     24475     FBgn0082989</span></span>
<span><span class="co">##   FBtr0091635       3L   1488906-1489045      + |     13780     FBgn0086670</span></span>
<span><span class="co">##   FBtr0113599       3L     261803-261953      - |     16973     FBgn0083014</span></span>
<span><span class="co">##   FBtr0113600       3L     831870-832008      - |     17070     FBgn0083057</span></span>
<span><span class="co">##                   tx_name</span></span>
<span><span class="co">##               &lt;character&gt;</span></span>
<span><span class="co">##       Newgene     Newgene</span></span>
<span><span class="co">##   FBtr0070129 FBtr0070129</span></span>
<span><span class="co">##   FBtr0070126 FBtr0070126</span></span>
<span><span class="co">##   FBtr0070128 FBtr0070128</span></span>
<span><span class="co">##   FBtr0070124 FBtr0070124</span></span>
<span><span class="co">##           ...         ...</span></span>
<span><span class="co">##   FBtr0114299 FBtr0114299</span></span>
<span><span class="co">##   FBtr0113582 FBtr0113582</span></span>
<span><span class="co">##   FBtr0091635 FBtr0091635</span></span>
<span><span class="co">##   FBtr0113599 FBtr0113599</span></span>
<span><span class="co">##   FBtr0113600 FBtr0113600</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Seqinfo object with 25 sequences from an unspecified genome; no seqlengths:</span></span>
<span><span class="co">##   seqnames             seqlengths isCircular genome</span></span>
<span><span class="co">##   2L                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   2R                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   3L                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   3R                         &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   4                          &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   ...                         ...        ...    ...</span></span>
<span><span class="co">##   211000022280481            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   211000022280494            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   211000022280703            &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   mitochondrion_genome       &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">##   rDNA                       &lt;NA&gt;       &lt;NA&gt;   &lt;NA&gt;</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="clear-linkedtxomes">Clear <em>linkedTxomes</em><a class="anchor" aria-label="anchor" href="#clear-linkedtxomes"></a>
</h2>
<p>The following code removes the entire table with information about
the <em>linkedTxomes</em>. This is just for demonstration, so that we
can show how to load a JSON file below.</p>
<p><strong>Note:</strong> Running this code will clear any information
about <em>linkedTxomes</em>. Don’t run this unless you really want to
clear this table!</p>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocFileCache</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">bfcloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTximetaBFC.html">getTximetaBFC</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="va">bfcloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="va">bfcloc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcinfo</a></span><span class="op">(</span><span class="va">bfc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">##   rid   rname create_time access_time rpath rtype fpath last_modified_time etag </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> BFC1  link… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> BFC2  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> BFC3  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> BFC4  exon… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> BFC5  gene… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> BFC6  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> BFC7  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: expires &lt;dbl&gt;</span></span></span></code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcremove</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcquery</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="st">"linkedTxomeTbl"</span><span class="op">)</span><span class="op">$</span><span class="va">rid</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcinfo</a></span><span class="op">(</span><span class="va">bfc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">##   rid   rname create_time access_time rpath rtype fpath last_modified_time etag </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> BFC2  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> BFC3  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> BFC4  exon… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> BFC5  gene… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> BFC6  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> BFC7  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: expires &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="loading-linkedtxome-json-files">Loading <em>linkedTxome</em> JSON files<a class="anchor" aria-label="anchor" href="#loading-linkedtxome-json-files"></a>
</h2>
<p>If a collaborator or the Suppmentary Files for a publication shares a
<code>linkedTxome</code> JSON file, we can likewise use
<code>tximeta</code> to automatically assemble the relevant metadata and
transcript ranges. This implies that the other person has used
<code>tximeta</code> with the function <code>makeLinkedTxome</code>
demonstrated above, pointing to their <em>Salmon</em> index and to the
FASTA and GTF source(s).</p>
<p>We point to the JSON file and use <code>loadLinkedTxome</code> and
then the relevant metadata is saved for persistent usage. In this case,
we saved the JSON file in a temporary directory.</p>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">jsonFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">indexDir</span><span class="op">)</span>, <span class="st">".json"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/linkedTxome.html">loadLinkedTxome</a></span><span class="op">(</span><span class="va">jsonFile</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## saving linkedTxome in bfc (first time)</span></span></code></pre>
<p>Again, using <code>tximeta</code> figures out whether it needs to
access the remote GTF or not, and assembles the appropriate object on
the user’s behalf.</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tximeta.html">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## importing quantifications</span></span></code></pre>
<pre><code><span><span class="co">## reading in files with read.delim (install 'readr' package for speed up)</span></span></code></pre>
<pre><code><span><span class="co">## 1 </span></span>
<span><span class="co">## found matching linked transcriptome:</span></span>
<span><span class="co">## [ LocalEnsembl - Drosophila melanogaster - release 98 ]</span></span>
<span><span class="co">## loading existing TxDb created: 2024-02-21 14:31:53</span></span>
<span><span class="co">## loading existing transcript ranges created: 2024-02-21 14:31:54</span></span></code></pre>
<pre><code><span><span class="co">## Warning in checkAssays2Txps(assays, txps): </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Warning: the annotation is missing some transcripts that were quantified.</span></span>
<span><span class="co">## 5 out of 33707 txps were missing from GTF/GFF but were in the indexed FASTA.</span></span>
<span><span class="co">## (This occurs sometimes with Ensembl txps on haplotype chromosomes.)</span></span>
<span><span class="co">## In order to build a ranged SummarizedExperiment, these txps were removed.</span></span>
<span><span class="co">## To keep these txps, and to skip adding ranges, use skipMeta=TRUE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Example missing txps: [FBtr0307759, FBtr0084079, FBtr0084080, ...]</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="clear-linkedtxomes-again">Clear <em>linkedTxomes</em> again<a class="anchor" aria-label="anchor" href="#clear-linkedtxomes-again"></a>
</h2>
<p>Finally, we clear the <em>linkedTxomes</em> table again so that the
above examples will work. This is just for the vignette code and not
part of a typical workflow.</p>
<p><strong>Note:</strong> Running this code will clear any information
about <em>linkedTxomes</em>. Don’t run this unless you really want to
clear this table!</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">bfcloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTximetaBFC.html">getTximetaBFC</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="va">bfcloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="va">bfcloc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcinfo</a></span><span class="op">(</span><span class="va">bfc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">##   rid   rname create_time access_time rpath rtype fpath last_modified_time etag </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> BFC2  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> BFC3  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> BFC4  exon… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> BFC5  gene… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> BFC6  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> BFC7  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> BFC8  link… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: expires &lt;dbl&gt;</span></span></span></code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcremove</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcquery</a></span><span class="op">(</span><span class="va">bfc</span>, <span class="st">"linkedTxomeTbl"</span><span class="op">)</span><span class="op">$</span><span class="va">rid</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcinfo</a></span><span class="op">(</span><span class="va">bfc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">##   rid   rname create_time access_time rpath rtype fpath last_modified_time etag </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> BFC2  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> BFC3  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> BFC4  exon… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> BFC5  gene… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> BFC6  Dros… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> BFC7  txpR… 2024-02-21… 2024-02-21… /tmp… rela… 1255…                 <span style="color: #BB0000;">NA</span> <span style="color: #BB0000;">NA</span>   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: expires &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="other-quantifiers">Other quantifiers<a class="anchor" aria-label="anchor" href="#other-quantifiers"></a>
</h2>
<p><code>tximeta</code> can import the output from any quantifiers that
are supported by <code>tximport</code>, and if these are not
<em>Salmon</em>, <em>alevin</em>, or <em>Sailfish</em> output, it will
simply return a non-ranged <em>SummarizedExperiment</em> by default.</p>
<p>An alternative solution is to wrap other quantifiers in workflows
that include metadata information JSON files along with each
quantification file. One can place these files in
<code>aux_info/meta_info.json</code> or any relative location specified
by <code>customMetaInfo</code>, for example
<code>customMetaInfo="meta_info.json"</code>. This JSON file is located
relative to the quantification file and should contain a tag
<code>index_seq_hash</code> with an associated value of the SHA-256 hash
of the reference transcripts. For computing the hash value of the
reference transcripts, see the <a href="https://github.com/COMBINE-lab/FastaDigest" class="external-link">FastaDigest</a> python
package. The hash value used by <em>Salmon</em> is the SHA-256 hash
value of the reference sequences stripped of the header lines, and
concatenated together with the empty string (so only cDNA sequences
combined without any new line characters). <em>FastaDigest</em> can be
installed with <code>pip install fasta_digest</code>.</p>
</div>
<div class="section level2">
<h2 id="automated-analysis-with-armor">Automated analysis with ARMOR<a class="anchor" aria-label="anchor" href="#automated-analysis-with-armor"></a>
</h2>
<p>This vignette described the use of <code>tximeta</code> to import
quantification data into R/Bioconductor with automatic detection and
addition of metadata. The <em>SummarizedExperiment</em> produced by
<code>tximeta</code> can then be provided to downstream statistical
analysis packages as described above. The <em>tximeta</em> package does
not contain any functionality for automated differential analysis.</p>
<p>The <a href="https://github.com/csoneson/ARMOR" class="external-link">ARMOR</a> workflow
does automate a variety of differential analyses, and make use of
<code>tximeta</code> for creation of a <em>SummarizedExperiment</em>
with attached annotation metadata. ARMOR stands for ``An Automated
Reproducible MOdular Workflow for Preprocessing and Differential
Analysis of RNA-seq Data’’ and is described in more detail in the
article by <span class="citation">Orjuela et al. (2019)</span>.</p>
</div>
<div class="section level2">
<h2 id="default-biocfilecahce-caching-location-update">Default BiocFileCahce Caching Location Update<a class="anchor" aria-label="anchor" href="#default-biocfilecahce-caching-location-update"></a>
</h2>
<p><em>tximeta</em> makes use of the default BiocFileCache location,
unless otherwise specified by the user. As of BiocFileCache version &gt;
1.15.1, the default caching location used by BiocFileCache has changed.
In order to continue to use the same cache (without re-downloading
files), please follow the steps in the BiocFileCache vignette, under the
heading <strong>Default Caching Location Update</strong>.</p>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>The development of <em>tximeta</em> has benefited from suggestions
from these and other individuals in the community:</p>
<ul>
<li>Vincent Carey</li>
<li>Lori Shepherd</li>
<li>Martin Morgan</li>
<li>Koen Van den Berge</li>
<li>Johannes Rainer</li>
<li>James Ashmore</li>
<li>Ben Johnson</li>
<li>Tim Triche</li>
<li>Kristoffer Vitting-Seerup</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: usethis</span></span></code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">##  os       Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">##  system   x86_64, linux-gnu</span></span>
<span><span class="co">##  ui       X11</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  en_US.UTF-8</span></span>
<span><span class="co">##  ctype    en_US.UTF-8</span></span>
<span><span class="co">##  tz       UTC</span></span>
<span><span class="co">##  date     2024-02-21</span></span>
<span><span class="co">##  pandoc   3.1.1 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package               </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version    </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  abind                    1.4-5       <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  AnnotationDbi          * 1.64.1      <span style="color: #555555;">2023-11-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationFilter         1.26.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationHub            3.10.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  Biobase                * 2.62.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocFileCache          * 2.10.1      <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocGenerics           * 0.48.1      <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocIO                   1.12.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocManager              1.30.22     <span style="color: #555555;">2023-08-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">##  BiocParallel             1.36.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocVersion              3.18.1      <span style="color: #555555;">2023-11-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biomaRt                  2.58.2      <span style="color: #555555;">2024-01-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  Biostrings               2.70.2      <span style="color: #555555;">2024-01-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  bit                      4.0.5       <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bit64                    4.0.5       <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bitops                   1.0-7       <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  blob                     1.2.4       <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bslib                    0.6.1       <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  cachem                   1.0.8       <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  cli                      3.6.2       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  codetools                0.2-19      <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.2)</span></span></span>
<span><span class="co">##  colorspace               2.1-0       <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  crayon                   1.5.2       <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  curl                     5.2.0       <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  DBI                      1.2.2       <span style="color: #555555;">2024-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  dbplyr                 * 2.4.0       <span style="color: #555555;">2023-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  DelayedArray             0.28.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  desc                     1.4.3       <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  DESeq2                 * 1.42.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  devtools               * 2.4.5       <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  digest                   0.6.34      <span style="color: #555555;">2024-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  dplyr                    1.1.4       <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  edgeR                  * 4.0.16      <span style="color: #555555;">2024-02-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  ellipsis                 0.3.2       <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  ensembldb                2.26.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  evaluate                 0.23        <span style="color: #555555;">2023-11-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  fansi                    1.0.6       <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  fastmap                  1.1.1       <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  filelock                 1.0.3       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  fishpond               * 2.8.0       <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  fs                       1.6.3       <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  generics                 0.1.3       <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  GenomeInfoDb           * 1.38.6      <span style="color: #555555;">2024-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  GenomeInfoDbData         1.2.11      <span style="color: #555555;">2024-02-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicAlignments        1.38.2      <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  GenomicFeatures        * 1.54.3      <span style="color: #555555;">2024-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  GenomicRanges          * 1.54.1      <span style="color: #555555;">2023-10-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ggplot2                  3.4.4       <span style="color: #555555;">2023-10-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  glue                     1.7.0       <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  gtable                   0.3.4       <span style="color: #555555;">2023-08-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  gtools                   3.9.5       <span style="color: #555555;">2023-11-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  highr                    0.10        <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  hms                      1.1.3       <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  htmltools                0.5.7       <span style="color: #555555;">2023-11-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  htmlwidgets              1.6.4       <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  httpuv                   1.6.14      <span style="color: #555555;">2024-01-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  httr                     1.4.7       <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  interactiveDisplayBase   1.40.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  IRanges                * 2.36.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  jquerylib                0.1.4       <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  jsonlite                 1.8.8       <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  KEGGREST                 1.42.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  knitr                    1.45        <span style="color: #555555;">2023-10-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  later                    1.3.2       <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  lattice                  0.22-5      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  lazyeval                 0.2.2       <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  lifecycle                1.0.4       <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  limma                  * 3.58.1      <span style="color: #555555;">2023-10-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  locfit                   1.5-9.8     <span style="color: #555555;">2023-06-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  magrittr                 2.0.3       <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Matrix                   1.6-5       <span style="color: #555555;">2024-01-11</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  MatrixGenerics         * 1.14.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  matrixStats            * 1.2.0       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  memoise                  2.0.1       <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  mime                     0.12        <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  miniUI                   0.1.1.1     <span style="color: #555555;">2018-05-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  munsell                  0.5.0       <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  org.Dm.eg.db           * 3.18.0      <span style="color: #555555;">2024-02-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  pillar                   1.9.0       <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgbuild                 1.4.3       <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgconfig                2.0.3       <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgdown                  2.0.7       <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgload                  1.3.4       <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  png                      0.1-8       <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  prettyunits              1.2.0       <span style="color: #555555;">2023-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  profvis                  0.3.8       <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  progress                 1.2.3       <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  promises                 1.2.1       <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  ProtGenerics             1.34.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  purrr                    1.0.2       <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  R6                       2.5.1       <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  ragg                     1.2.7       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rappdirs                 0.3.3       <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Rcpp                     1.0.12      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  RCurl                    1.98-1.14   <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  remotes                  <span style="color: #BB00BB; font-weight: bold;">2.4.2.9000 </span> <span style="color: #555555;">2024-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/remotes@6fe8836)</span></span></span>
<span><span class="co">##  restfulr                 0.0.15      <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.2)</span></span></span>
<span><span class="co">##  rjson                    0.2.21      <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rlang                    1.1.3       <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rmarkdown                2.25        <span style="color: #555555;">2023-09-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Rsamtools                2.18.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RSQLite                  2.3.5       <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rtracklayer              1.62.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Arrays                 1.2.0       <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Vectors              * 0.40.2      <span style="color: #555555;">2023-11-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  sass                     0.4.8       <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  scales                   1.3.0       <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  sessioninfo              1.2.2       <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  shiny                    1.8.0       <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  SingleCellExperiment     1.24.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  SparseArray              1.2.4       <span style="color: #555555;">2024-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.18 (R 4.3.2)</span></span></span>
<span><span class="co">##  statmod                  1.5.0       <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  stringi                  1.8.3       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  stringr                  1.5.1       <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  SummarizedExperiment   * 1.32.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  svMisc                   1.2.3       <span style="color: #555555;">2021-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  systemfonts              1.0.5       <span style="color: #555555;">2023-10-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  textshaping              0.3.7       <span style="color: #555555;">2023-10-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  tibble                   3.2.1       <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  tidyselect               1.2.0       <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  tximeta                * 1.21.4      <span style="color: #555555;">2024-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  tximport                 1.30.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  urlchecker               1.0.1       <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  usethis                * 2.2.2       <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  utf8                     1.2.4       <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  vctrs                    0.6.5       <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  withr                    3.0.0       <span style="color: #555555;">2024-01-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  xfun                     0.42        <span style="color: #555555;">2024-02-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  XML                      3.99-0.16.1 <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  xml2                     1.3.6       <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  xtable                   1.8-4       <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  XVector                  0.42.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  yaml                     2.3.8       <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  zlibbioc                 1.48.0      <span style="color: #555555;">2023-10-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-granges" class="csl-entry">
Lawrence, Wolfgang AND Pagès, Michael AND Huber. 2013. <span>“<span class="nocase">Software for Computing and Annotating Genomic
Ranges</span>.”</span> <em>PLOS Computational Biology</em> 9 (8): 1–10.
</div>
<div id="ref-Love2020" class="csl-entry">
Love, Michael I., Charlotte Soneson, Peter F. Hickey, Lisa K. Johnson,
N. Tessa Pierce, Lori Shepherd, Martin Morgan, and Rob Patro. 2020.
<span>“<span class="nocase">Tximeta: Reference sequence checksums for
provenance identification in RNA-seq</span>.”</span> <em>PLOS
Computational Biology</em> 16: e1007664. <a href="https://doi.org/10.1371/journal.pcbi.1007664" class="external-link">https://doi.org/10.1371/journal.pcbi.1007664</a>.
</div>
<div id="ref-Orjuelag2019" class="csl-entry">
Orjuela, Stephany, Ruizhu Huang, Katharina M. Hembach, Mark D. Robinson,
and Charlotte Soneson. 2019. <span>“<span class="nocase">ARMOR: An
Automated Reproducible MOdular Workflow for Preprocessing and
Differential Analysis of RNA-seq Data</span>.”</span> <em>G3: Genes,
Genomes, Genetics</em>.
</div>
<div id="ref-Patro2017" class="csl-entry">
Patro, Rob, Geet Duggal, Michael I. Love, Rafael A. Irizarry, and Carl
Kingsford. 2017. <span>“Salmon Provides Fast and Bias-Aware
Quantification of Transcript Expression.”</span> <em>Nature
Methods</em>. <a href="https://doi.org/10.1038/nmeth.4197" class="external-link">https://doi.org/10.1038/nmeth.4197</a>.
</div>
<div id="ref-ensembldb" class="csl-entry">
Rainer, Johannes, Laurent Gatto, and Christian X Weichenberger. 2019.
<span>“<span class="nocase">ensembldb: an R package to create and use
Ensembl-based annotation resources</span>.”</span>
<em>Bioinformatics</em>, January.
</div>
<div id="ref-Soneson2015" class="csl-entry">
Soneson, Charlotte, Michael I. Love, and Mark Robinson. 2015.
<span>“<span class="nocase">Differential analyses for RNA-seq:
transcript-level estimates improve gene-level inferences</span>.”</span>
<em>F1000Research</em> 4. <a href="https://doi.org/10.12688/f1000research.7563.1" class="external-link">https://doi.org/10.12688/f1000research.7563.1</a>.
</div>
<div id="ref-Srivastava2019" class="csl-entry">
Srivastava, Avi, Laraib Malik, Tom Sean Smith, Ian Sudbery, and Rob
Patro. 2019. <span>“<span class="nocase">Alevin efficiently estimates
accurate gene abundances from dscRNA-seq data</span>.”</span> <em>Genome
Biology</em> 20 (65). <a href="https://doi.org/10.1186/s13059-019-1670-y" class="external-link">https://doi.org/10.1186/s13059-019-1670-y</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://mikelove.github.io" class="external-link">Michael Love</a>, <a href="https://csoneson.github.io/" class="external-link">Charlotte Soneson</a>, <a href="https://www.peterhickey.org/" class="external-link">Peter Hickey</a>, <a href="https://combine-lab.github.io/" class="external-link">Rob Patro</a>, NIH NHGRI, CZI.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
